---
 src/Makefile.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -88,7 +88,7 @@
 @BUILD_STATIC_TRUE@am__DEPENDENCIES_1 = ../libdw/libdw.a $(libelf) \
 @BUILD_STATIC_TRUE@	$(libebl)
 am__DEPENDENCIES_2 =
-addr2line_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_2)
+addr2line_DEPENDENCIES = $(am__DEPENDENCIES_1) $(libelf) $(am__DEPENDENCIES_2)
 ar_SOURCES = ar.c
 ar_OBJECTS = ar.$(OBJEXT)
 ar_DEPENDENCIES = libar.a $(libelf) $(libeu) $(am__DEPENDENCIES_2)
@@ -346,7 +346,7 @@
 ld_LDFLAGS = -rdynamic
 elflint_LDADD = $(libebl) $(libelf) $(libeu) $(libmudflap) -ldl
 findtextrel_LDADD = $(libdw) $(libelf) $(libmudflap)
-addr2line_LDADD = $(libdw) $(libmudflap)
+addr2line_LDADD = $(libdw) $(libelf) $(libmudflap)
 elfcmp_LDADD = $(libebl) $(libelf) $(libmudflap) -ldl
 objdump_LDADD = $(libasm) $(libebl) $(libelf) $(libeu) $(libmudflap) -ldl
 ranlib_LDADD = libar.a $(libelf) $(libeu) $(libmudflap)
