---
 term-utils/script.c |   14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

--- a/term-utils/script.c
+++ b/term-utils/script.c
@@ -330,11 +330,11 @@ finish(int dummy __attribute__ ((__unuse
 	pid_t pid;
 	int errsv = errno;
 
-	while ((pid = wait3(&status, WNOHANG, 0)) > 0)
-		if (pid == child) {
-			childstatus = status;
+	while ((pid = waitpid(-1, &status, WNOHANG)) > 0) {
+		if (pid == child)
 			die = 1;
-		}
+		childstatus = status;
+	}
 
 	errno = errsv;
 }
@@ -508,7 +508,11 @@ done(void) {
 		else
 			exit(WEXITSTATUS(childstatus));
 	}
-	exit(EXIT_SUCCESS);
+
+	if (childstatus < 256 && childstatus > 0)
+		exit(childstatus);
+
+	exit(childstatus>>8);
 }
 
 void
