---
 term-utils/script.c |   24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

--- a/term-utils/script.c
+++ b/term-utils/script.c
@@ -108,11 +108,15 @@ int	fflg = 0;
 int	qflg = 0;
 int	tflg = 0;
 int	rflg = 0;
+int	iflg = 0;
 int	forceflg = 0;
 
 int die;
 int resized;
 
+fd_set rds;
+fd_set master_rds;
+
 static void
 die_if_link(char *fn) {
 	struct stat s;
@@ -184,7 +188,7 @@ main(int argc, char **argv) {
 	textdomain(PACKAGE);
 	atexit(close_stdout);
 
-	while ((ch = getopt_long(argc, argv, "ac:efqrt::Vh", longopts, NULL)) != -1)
+	while ((ch = getopt_long(argc, argv, "ac:efiqrt::Vh", longopts, NULL)) != -1)
 		switch(ch) {
 		case 'a':
 			aflg = 1;
@@ -195,6 +199,9 @@ main(int argc, char **argv) {
 		case 'e':
 			eflg = 1;
 			break;
+		case 'i':
+			iflg++;
+			break;
 		case 'f':
 			fflg = 1;
 			break;
@@ -239,6 +246,14 @@ main(int argc, char **argv) {
 		fail();
 	}
 
+	if (iflg) {
+		sigemptyset(&sa.sa_mask);
+		sa.sa_flags = 0;
+
+		sa.sa_handler = SIG_IGN;
+		sigaction(SIGTTOU, &sa, NULL);
+	}
+
 	shell = getenv("SHELL");
 	if (shell == NULL)
 		shell = _PATH_BSHELL;
@@ -300,11 +315,18 @@ void
 doinput(void) {
 	ssize_t cc;
 	char ibuf[BUFSIZ];
+	int select_ret;
 
 	if (close_stream(fscript) != 0)
 		errx(EXIT_FAILURE, _("write error"));
 
+	FD_ZERO(&master_rds);
+	FD_SET(0, &master_rds);
 	while (die == 0) {
+		memcpy(&rds, &master_rds, sizeof(master_rds));
+		select_ret = select(1, &rds, NULL, NULL, NULL);
+		if (select_ret < 0)
+			break;
 		if ((cc = read(STDIN_FILENO, ibuf, BUFSIZ)) > 0) {
 			ssize_t wrt = write(master, ibuf, cc);
 			if (wrt < 0) {
