From 1784a3ed155fa287c708a50182c0b4e836aab459 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Thu, 6 Dec 2012 10:55:58 +0800
Subject: [PATCH] Do not strip the acpidump executable.

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 acpidump/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/acpidump/Makefile b/acpidump/Makefile
index 7fcde3c..22f74e5 100644
--- a/acpidump/Makefile
+++ b/acpidump/Makefile
@@ -1,7 +1,7 @@
 PROG= acpidump
 SRCS=	acpidump.c
 KERNEL_INCLUDE := ../include
-CFLAGS += -Wall -Wstrict-prototypes -Wdeclaration-after-statement -Os -s -D_LINUX -DDEFINE_ALTERNATE_TYPES -I$(KERNEL_INCLUDE) 
+CFLAGS += -Wall -Wstrict-prototypes -Wdeclaration-after-statement -D_LINUX -DDEFINE_ALTERNATE_TYPES -I$(KERNEL_INCLUDE) 
 
 all: acpidump
 $(PROG) : $(SRCS)
-- 
1.7.4.1

