Subject: [PATCH] Makefile: fix to build madt

Upstream-Status: Inappropriate [dead upstream]

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 Makefile      |  2 +-
 madt/Makefile | 15 ++++++++++++---
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 3bea2f6..2c4e99f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-DIRS = acpidump acpixtract turbostat
+DIRS = acpidump acpixtract turbostat madt
 
 all:
 	for i in $(DIRS); do $(MAKE) -C $$i $@; done
diff --git a/madt/Makefile b/madt/Makefile
index 1bcd4bb..725858f 100644
--- a/madt/Makefile
+++ b/madt/Makefile
@@ -1,6 +1,15 @@
+PROG= madt
+SRCS= madt.c
+
+all: $(PROG)
+$(PROG): $(SRCS)
+	$(CC) $(CFLAGS) $(SRCS) -o $(PROG)
+
 test: madt
 	./madt < APIC
-madt: madt.c acpi.h tables.c
-	cc -o madt madt.c
+	./madt < APIC.bad
+
+CLEANFILES= $(PROG)
+
 clean:
-	rm madt
+	rm -f $(CLEANFILES) $(patsubst %.c,%.o, $(SRCS)) *~
-- 
2.0.0

